tensorflow_datasets/scripts/documentation/dataset_markdown_builder.py

```diff
@@ -0,0 +1,412 @@
+"""Dataset catalog documentation template.
+
+Displayed in https://www.tensorflow.org/datasets/catalog/.
+
+"""
+import dataclasses
+import functools
+import textwrap
+from typing import Callable, List
+
+import tensorflow_datasets as tfds
+
+
+@dataclasses.dataclass
+class Section(object):
+  get_signature: Callable[[tfds.core.DatasetBuilder], str]
+  make: Callable[[tfds.core.DatasetBuilder], str]
+
+# --------------------------- Builder sections ---------------------------
+
+
+def display_description(builder):
+  return textwrap.dedent(
+      f"""\
+      *   **Description**:
+
+      """
+  ) + tfds.core.utils.dedent(builder.info.description) + "\n"
+
+
+def display_config_description(builder):
+  if builder.builder_config:
+    return textwrap.dedent(
+        f"""\
+        *   **Config description**: {builder.builder_config.description}
+        """
+    )
+  return ""
+
+
+def display_homepage(builder):
+  return textwrap.dedent(
```

Comment:
Couldn't dedent be applied automatically without having to specify it in each functions ?